umbraco/Umbraco.CMS.Backoffice
# Add localization files

Adds localization files from the existing Umb v13 repo, updated into the new TS/JSON format. 

There's a unique commit for each localization to make it easier to cherry-pick/reject. I've not migrated the two existing files `da-dk.ts` and `en-us.ts`.

## Description

I wrote a [console app](https://github.com/liamlaverty/umb-localization-migrator/) which takes the [existing XML localization files in the v13 repo](https://github.com/umbraco/Umbraco-CMS/tree/v13/contrib/src/Umbraco.Core/EmbeddedResources/Lang) and migrates them into the new typescript format outlined in the Storybook's [localization section](https://ambitious-stone-0033b3603.1.azurestaticapps.net/?path=/docs/api-localization-intro--docs). 

This technique will miss a lot of properties. Any new features added into v14, or any features where the localization alias has changed will unfortunately not be migrated.  To help with this, in the console app's repo, there's a record of the properties which haven't been migrated for each language in the [reports folder](https://github.com/liamlaverty/umb-localization-migrator/tree/main/IOFiles/DiffFinder/reports). For example, if a property existed in the V14 en-US file, but didn't exist in the V13 fr-fr file, that is detailed in [fr-fr-migration-report.md](https://github.com/liamlaverty/umb-localization-migrator/blob/main/IOFiles/DiffFinder/reports/fr-fr-migration-report.md).

## Types of changes



- [ ✅ ] New feature (non-breaking change which adds functionality)
- [  ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Motivation and context

V13 has a lot of localization files, which are currently not ported to V14

## How to test?

- Change the back office language to fr-fr
- Attempt to copy any entity in the backoffice
- The text should read "Copier" instead of "Copy"


## Screenshots (if appropriate)

## Checklist



- [ ✅ ] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/main/.github/CONTRIBUTING.md)>)** document.
